python/ray/train/tests/test_worker_group.py

```diff
@@ -59,6 +76,46 @@ def test_worker_restart(ray_start_2_cpus):
     wg.execute(lambda: 1)
 
 
+def test_worker_with_gpu_ids(ray_start_2_cpus_and_gpus):
+    num_gpus = 2
+    wg = WorkerGroup(num_workers=2, num_gpus_per_worker=1)
+    assert len(wg.workers) == 2
+    time.sleep(1)
+    assert ray_constants.GPU not in ray.available_resources()
+    wg.execute(lambda: 1)
+    assert len(wg.workers) == 2
+    for w in wg.workers:
+        gpu_and_accelerator_ids = w.metadata.resource_ids
+        # 2_CPUs, 1_GPU
+        assert len(gpu_and_accelerator_ids) == 3
```

Comment:
It's unclear to me what the purpose of this comment is. Also I'm not sure if we should assert the length of this, as it may change if we add more resource types in the future?

Reply:
List doesn't change because these resources are configured at test/UT level (let say ray_start_2_cpus_and_gpus). Comments were added during my UT debugging.

Reply:
Oh what is the output of this? I tried running a super naive script on my laptop that prints out `ray.get_runtime_context().get_resource_ids()`, and it showed up as `{'GPU': [], 'neuron_cores': [], 'TPU': []}`.

Reply:
You are right, i was under the impression (got lost with refactors) I get CPU/GPU/neuron_cores. It's actually TPU which got added recently.  Fixed the UTs by removing length check